pyvo/registry/regtap.py

```diff
@@ -399,10 +400,22 @@ def __init__(self, access_url, *, standard_id=None,
         else:
             self.is_vosi = False
 
+        # a service is user visible if it has a corresponding service class
+        if self.standard_id is not None and self.standard_id != "":
```

Comment:
I think there is no reason to ward against empty standard_id here (in addition to None, I mean).

Reply:
The remove data document tests fail without it. Looks like some non standard services have an empty string in self.standard_id. I did not investigate further than adding this condition.